backend/leaderboard/views.py

```
@@ -904,56 +904,125 @@ def trending(self, request):
 
         from datetime import timedelta
         from users.models import User
 
         cutoff = timezone.now() - timedelta(days=30)
 
-        # Aggregate points per user from last 30 days
-        trending_users = list(
-            Contribution.objects.filter(
-                created_at__gte=cutoff,
-                user__visible=True,
+        def build_points_query(recent_only=True):
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `build_points_query`

(ANN202)